app/src/main/java/com/kms/cura/view/activity/PatientHealthDetailActivity.java

```diff
@@ -0,0 +1,304 @@
+        pastHealthEntities = new ArrayList<>();
+        for (int i = 0; i < entities.size(); ++i) {
+            if (entities.get(i).isPastHealth()) {
+                pastHealthEntities.add(entities.get(i));
+            } else {
+                curHealthEntities.add(entities.get(i));
+            }
+        }
+    }
+
+    private void setUpToolbar() {
+        toolbar = (Toolbar) findViewById(R.id.tbHealthDetail);
+        toolbar.getMenu().clear();
+        tvTitle = (TextView) toolbar.findViewById(R.id.tvHealthTitle);
+        btnBack = (ImageButton) toolbar.findViewById(R.id.btnDetailBack);
+        if (healthEntity.isCondition()) {
+            tvTitle.setText(getString(R.string.condition));
+        } else {
+            tvTitle.setText(getString(R.string.symptom));
+        }
+        toolbar.inflateMenu(R.menu.menu_edit);
+        toolbar.setOnMenuItemClickListener(this);
+        btnBack.setOnClickListener(this);
+    }
+
+    private void modifyToolbar() {
+        toolbar.getMenu().clear();
+        if (edit_mode) {
+            toolbar.inflateMenu(R.menu.menu_done);
+        } else {
+            toolbar.inflateMenu(R.menu.menu_edit);
+        }
+    }
+
+    private void changeMode() {
+        edit_mode = !edit_mode;
+        setEditVisibility();
+        modifyToolbar();
+    }
+
+    private void setEditVisibility() {
+        if (edit_mode) {
+            commentLayout.removeAllViews();
+            View view = getLayoutInflater().inflate(R.layout.comment_edit_text, commentLayout, true);
+            edtComment = (EditText) view.findViewById(R.id.edtComment);
+            edtComment.setText(healthEntity.getComment());
+            btnEnd.setVisibility(View.VISIBLE);
+            btnStart.setVisibility(View.VISIBLE);
+            tvEndDate.setClickable(true);
+            tvStartDate.setClickable(true);
+        } else {
+            commentLayout.removeAllViews();
+            View view = getLayoutInflater().inflate(R.layout.comment_text_view, commentLayout, true);
+            tvComment = (TextView) view.findViewById(R.id.tvComment);
+            tvComment.setText(healthEntity.getComment());
+            btnEnd.setVisibility(View.INVISIBLE);
+            btnStart.setVisibility(View.INVISIBLE);
+            tvEndDate.setClickable(false);
+            tvStartDate.setClickable(false);
+        }
+    }
+
+
+    @Override
+    public boolean onMenuItemClick(MenuItem item) {
+        if (item.getItemId() == R.id.btnEdit) {
+            changeMode();
+        }
+        if (item.getItemId() == R.id.btnDone) {
+            // make change in database and refresh CurrentUser
+            healthEntity.setComment(edtComment.getText().toString());
+            healthEntity.setStartDate(toDateSQL(startYear, startMonth, startDay));
+            if (endDateSet) {
+                healthEntity.setEndDate(toDateSQL(endYear, endMonth, endDay));
+            }
+            setDataOnView(healthEntity);
+            changeMode();
```

Comment:
Also this block of code should be in a method name something like doneButtonPressed